P202-Training.Domain/UsersService.cs

```diff
@@ -0,0 +1,55 @@
+﻿using System;
+using System.Collections.Generic;
+using AutoMapper;
+using P202.Training.Data;
+using P202.Training.Data.Entities;
+using P202.Training.Data.Repositories;
+using P202.Training.Domain.Models;
+
+namespace P202.Training.Domain
+{
+    public class UsersService : IUsersService
+    {
+        public void CreateUser(Users users)
+        {
+            if (users == null) return;
+            Mapper.Initialize(cfg => cfg.CreateMap<Users, User>());
+            var mapUser = Mapper.Map<User>(users);
+            var userRepository = new UserRepository(new NHibernateSessionManager());
+            userRepository.CreateUser(mapUser);
+        }
+
+        public Users DeleteUser(int userId)
+        {
+            return new Users();
+        }
+
+        public IList<Users> ListUsers()
+        {
+            var userRepository = new UserRepository(new NHibernateSessionManager());
+            var listUsers= userRepository.GetAllUsers();
+
+            Mapper.Initialize(cfg => cfg.CreateMap<IList<User>, IList<Users>>());
+            var mapUser = Mapper.Map<IList<Users>>(listUsers);
+            return mapUser;
+        }
+
+        public Users ReadUser(int userId)
+        {
+            return new Users()
```

Comment:
If you're gonna implement this method in a separated branch it's OK, but please leave a TODO comment